orm/iterator_property_test.go

```diff
@@ -0,0 +1,82 @@
+package orm_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/orm/testdata"
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+)
+
+func TestPaginationProperty(t *testing.T) {
+	t.Run("TestPagination", rapid.MakeCheck(func(t *rapid.T) {
+		// Create a slice of group members
+		groupMembers := rapid.SliceOf(genGroupMember).Draw(t, "groupMembers").([]*testdata.GroupMember)
+
+		// Choose a random limit for paging
+		upperLimit := uint64(len(groupMembers))
+		if upperLimit == 0 {
+			upperLimit = 1
+		}
+		limit := rapid.Uint64Range(1, upperLimit).Draw(t, "limit").(uint64)
+
+		// Reconstruct the slice from the pages
+		reconstructedGroupMembers := make([]*testdata.GroupMember, 0, len(groupMembers))
+		for offset := uint64(0); offset < uint64(len(groupMembers)); offset += limit {
+			pageRequest := &query.PageRequest{
+				Key:        nil,
```

Comment:
would that make sense to also test this using `Key`?

Reply:
Done https://github.com/regen-network/regen-ledger/pull/493/commits/fc946a0163b27722a9e1c7b9f76dd005ca44d613